lib/common/strings.c

```diff
@@ -111,6 +111,39 @@ pcmk__scan_ll(const char *text, long long *result, long long default_value)
     return rc;
 }
 
+/*!
+ * \internal
+ * \brief Scan an integer value from a string, constrained to a minimum
+ *
+ * \param[in]  text           The string to scan (may be NULL)
+ * \param[out] result         Where to store result (or NULL to ignore)
+ * \param[in]  minimum        Value to use as default and minimum
+ *
+ * \return Standard Pacemaker return code
+ */
+int
+pcmk__scan_min_int(const char *text, int *result, int minimum)
+{
+    int rc;
+    long long result_ll;
+
+    rc = pcmk__scan_ll(text, &result_ll, (long long) minimum);
+
+    if (result_ll < (long long) minimum) {
+        crm_warn("Clipped '%s' to minimum acceptable value %d", text, minimum);
+        result_ll = (long long) minimum;
+
+    } else if (result_ll > INT_MAX) {
+        crm_warn("Clipped '%s' to maximum integer %d", text, INT_MAX);
+        result_ll = (long long) INT_MAX;
```

Comment:
It seems like this might be worth a mention in the doc block.

Reply:
Done. It will now return EOVERFLOW in that case as well, for consistency with pcmk__scan_ll().